pgmpy/base/DAG.py

```
@@ -1163,6 +1168,19 @@ def to_daft(
         ...             node_params={'a': {'shape': 'rectangle'}})
         <daft.PGM at 0x7f9bb48b0bb0>
         """
+        # Validate edge strengths if plotting is requested
+        if plot_edge_strength:
+            missing_strengths = []
+            for u, v in self.edges():
+                if "strength" not in self.edges[(u, v)]:
+                    missing_strengths.append((u, v))
+
+            if missing_strengths:
+                logger.warning(
```

We should throw an error here instead of a warning.